### ISSUE TITLE
Code formatting option to allow space before semicolon

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#34557]
<p>I use C#, but this could apply in other languages too.</p><p>Please create a code formatting option to allow an optional space <strong>before</strong> semicolons at the end of statements. For example, I want statements to look like this ...</p><pre>int x = 0 ;</pre><p>... <strong>not</strong> like this ...</p><pre>int x = 0;</pre><p>Of course, if there is a way to do this already, most grateful for advice on that.</p><p>Thanks!</p>

_This issue has been moved from https://developercommunity.visualstudio.com/content/idea/506639/code-formatting-option-to-allow-space-before-semic.html
VSTS ticketId: 829830_
_These are the original issue comments:_

Jane Wu [MSFT] on 3/27/2019, 00:35 AM (35 hours ago): <p>Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further.&nbsp;We'll provide an update once the issue has been triaged by the product team.</p>

**Comment by CyrusNajmabadi** [comment]
This doesn't seem to be a style i've ever seen *anywhere* it the C#/.net ecosystem.  I don't think it would be a good idea to provide this.  Perhaps it could be something a 3rd party could do as some part of a general 'formatting extensibility' work item.

**Comment by sharwell** [comment]
Design review conclusion: This is a candidate for the formatter extensibility API we'd like to add. We're leaving the issue open so we can validate the extensibility API supports this scenario, but marking it as blocked on it.

**Comment by UnitedMarsupials** [comment]
Meanwhile, how about an option to _explicitly remove_ such spaces? I've inherited code, that uses them extensively -- and would like the removal of such blanks to be handled by astyle...

**Comment by sharwell** [comment]
@UnitedMarsupials That behavior should already be implemented in the formatter. If you are not seeing it, can you file a new issue with the details?

**Comment by CyrusNajmabadi** [comment]
Closing out as we are not planning on doing this ourselves.  This would fall under external extensibility tracked by https://github.com/dotnet/roslyn/issues/31691.
